doc/ref/modules/index.rst

```diff
@@ -141,33 +141,34 @@ This will ensure that the text outputter is used.
 Virtual Modules
 ===============
 
-Sometimes an execution module should be presented in a generic way. A good example of this
-can be found in the package manager modules. The package manager changes from
-one operating system to another, but the Salt execution module that interfaces with the
-package manager can be presented in a generic way.
-
-The Salt modules for package managers all contain a ``__virtual__`` function
-which is called to define what systems the module should be loaded on.
-
-The ``__virtual__`` function is used to return either a
-:ref:`string <python2:typesseq>` or :py:data:`False`. If
-False is returned then the module is not loaded, if a string is returned then
-the module is loaded with the name of the string.
-
-This means that the package manager modules can be presented as the ``pkg`` module
-regardless of what the actual module is named.
+Virtual modules let you override the name of a module in order to use the same
+name to refer to one of several similar modules. The specific module that is
+loaded for a virtual name is selected based on the current platform or
+environment.
+
+For example, packages are managed across platforms using the ``pkg`` module.
+``pkg`` is a virtual module name that is
+an alias for the specific package manager module that is loaded on a specific
+system (for example, :mod:`yumpkg <salt.modules.yumpkg>` on RHEL/CentOS systems
+, and :mod:`aptpkg <salt.modules.aptpkg>` on Ubuntu).
+
+Virtual module names are set using the ``__virtual__`` function and the
+:ref:`virtual name <modules-virtual-name>`.
+
+``__virtual__`` Function
+========================
+
+The ``__virtual__`` function
+returns either a :ref:`string <python2:typesseq>`, :py:data:`True`, or
+:py:data:`False`. If a string is returned then the module is loaded using the
+name of the string as the virtual name. If ``True`` is returned the module is
+loaded using the current module name. If ``False`` is returned the module is not loaded.
+``False`` lets the module perform system checks and prevent loading if dependencies
+are not met.
```

Comment:
@jacobhammons, the `__virtual__` function returns either a single argument, the virtualname, `True`, `False` or two arguments, `False`, and a string describing the reason the module was not loaded.